test/factories/operatingsystem.rb

```diff
@@ -3,8 +3,43 @@
     sequence(:name) { |n| "operatingsystem#{n}" }
     sequence(:major) { |n| n }
 
+    trait :with_os_defaults do
+      after(:create) do |os,evaluator|
+        os.config_templates.each do |tmpl|
+          FactoryGirl.create(:os_default_template,
+                             :operatingsystem => os,
+                             :config_template => tmpl,
+                             :template_kind => tmpl.template_kind)
+        end
+      end
+    end
+
+    trait :with_provision do
+      config_templates {
+        [FactoryGirl.create(:config_template, :template_kind => TemplateKind.find_by_name('provision'))]
+      }
+      with_os_defaults
+    end
+
+    trait :with_archs do
+      architectures { [FactoryGirl.create(:architecture)] }
+    end
+
+    trait :with_media do
+      media { [FactoryGirl.create(:medium)] }
+    end
+
+    trait :with_ptables do
+      ptables { [FactoryGirl.create(:ptable)] }
+    end
+
+    trait :with_associations do
+      with_archs
+      with_media
+      with_ptables
+    end
+
     factory :coreos, class: Coreos do
-      association :media, :architecture, :ptables
```

Comment:
These simply don't work, it only works on a 1-1 relationship, not 1-many.  None of these factories were actually used until I attempted to use debian7_0.